File: Shosta.Functions/Domain/Entities/Organisation/Organisation.cs
using System.ComponentModel.DataAnnotations;

namespace Shosta.Functions.Domain.Entities.Organisation;

public class Organisation
{
    public int Id { get; set; }
    public int Year { get; set; }
    
    [MaxLength(255)] public string? BandPicture { get; set; }
    [MaxLength(255)] public string? BandTitle { get; set; }
    [MaxLength(4000)] public string? BandPresentation { get; set; }
    [MaxLength(255)] public string? CommitteePicture { get; set; }
    [MaxLength(255)] public string? CommitteeTitle { get; set; }
    [MaxLength(4000)] public string? CommitteePresentation { get; set; }
    
    public virtual IList<CommitteeMember>? CommitteeMembers { get; set; }
    public virtual IList<Sponsor>? Sponsors { get; set; }
}

File: UI/src/app/about/api/organisation.ts
export interface OrganisationContainer {
  Value: Organisation;
  StatusCode: number;
}

export interface Organisation {
  Year: number;
  WelcomeText: string;
  ContactPersonText: string;
  BandPicture?: string;
  BandTitle?: string;
  BandPresentation?: string;
  CommitteePicture?: string;
  CommitteeTitle?: string;
  CommitteePresentation?: string;
  CommitteeMembers?: CommitteeMember[];
}

export interface CommitteeMember {
  Function?: string;
  FirstName?: string;
  LastName?: string;
  Address?: string;
  Zip?: string;
  City?: string;
  PhoneNumber?: string;
  Email?: string;
  Presentation?: string;
  Picture?: string;
  IsContactPerson: boolean;
}

export interface Sponsor {
  filename: string;
  alt: string;
}

export interface EmailData {
  RecaptchaResponse?: string;
  FirstName?: string;
  LastName?: string;
  Email?: string;
  Phone?: string;
  Message?: string;
}

export interface EmailSendResult {
  success: boolean;
  message?: string;
}
